index/scorch/persister.go

```diff
@@ -1283,15 +1293,15 @@ func (s *Scorch) removeOldZapFiles() error {
 // duration. This results in all of them being purged from the boltDB
 // and the next iteration of the removeOldData() would end up protecting
 // latest contiguous snapshot which is a poor pattern in the rollback checkpoints.
-// Hence we try to retain atleast retentionFactor portion worth of old snapshots
+// Hence we try to retain atmost retentionFactor portion worth of old snapshots
 // in such a scenario using the following function
 func getBoundaryCheckPoint(retentionFactor float64,
 	checkPoints []*snapshotMetaData, timeStamp time.Time,
 ) time.Time {
 	if checkPoints != nil {
 		boundary := checkPoints[int(math.Floor(float64(len(checkPoints))*
 			retentionFactor))]
-		if timeStamp.Sub(boundary.timeStamp) < 0 {
+		if timeStamp.Sub(boundary.timeStamp) > 0 {
 			// too less checkPoints would be left.
```

Comment:
I think I follow the code here, but would you rephrase this comment here as well - indicating that we're moving the boundary on account of no checkpoints within?

Reply:
done thansk!